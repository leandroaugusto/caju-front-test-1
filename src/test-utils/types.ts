export interface ITestUtils {
  children?: React.ReactNode
}
