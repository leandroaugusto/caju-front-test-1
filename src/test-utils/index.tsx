import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";

import { ITestUtils } from "./types";

import theme from "~/theme";

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
