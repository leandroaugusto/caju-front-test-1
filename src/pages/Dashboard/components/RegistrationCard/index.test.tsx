import { customRender, screen } from "~/test-utils";

import { RegistrationCard } from ".";
import {
  ERegistrationsStatus,
  TRegistrationsData,
} from "~/types/registrations.types";

const registrationsMock: TRegistrationsData = {
  admissionDate: "22/10/2023",
  email: "luiz@caju.com.br",
  employeeName: "Luiz Filho",
  status: ERegistrationsStatus.APPROVED,
  cpf: "56642105087",
  id: "3",
};

describe("RegistrationCard", () => {
  it("Should show RegistrationCard component", () => {
    const { container } = customRender(
      <RegistrationCard data={registrationsMock} />
    );

    const employeeNameContainer = screen.getByTestId("employee-name");
    const employeeEmailContainer = screen.getByTestId("employee-email");
    const employeeAdmissionDateContainer = screen.getByTestId(
      "employee-admission-date"
    );
    const reviewButton = screen.getByTestId("review-button");

    expect(employeeNameContainer).toHaveTextContent(
      registrationsMock.employeeName
    );
    expect(employeeEmailContainer).toHaveTextContent(registrationsMock.email);
    expect(employeeAdmissionDateContainer).toHaveTextContent(
      registrationsMock.admissionDate
    );
    expect(reviewButton).toHaveTextContent("Revisar novamente");

    expect(container).toMatchSnapshot();
  });

  it("Should show RegistrationCard with review status", () => {
    registrationsMock.status = ERegistrationsStatus.REVIEW;

    const { container } = customRender(
      <RegistrationCard data={registrationsMock} />
    );

    const approvedButton = screen.getByTestId("approved-button");
    const reproveButton = screen.getByTestId("reproved-button");

    expect(approvedButton).toHaveTextContent("Aprovar");
    expect(reproveButton).toHaveTextContent("Reprovar");

    expect(container).toMatchSnapshot();
  });
});
