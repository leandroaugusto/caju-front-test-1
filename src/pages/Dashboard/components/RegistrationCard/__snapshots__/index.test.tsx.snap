// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationCard Should show RegistrationCard component 1`] = `
<div>
  <div
    class="sc-elDIKY gPXLze"
    data-testid="card-APPROVED-status"
  >
    <div
      class="sc-fQpRED hGgTXR"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <h3
        data-testid="employee-name"
      >
        Luiz Filho
      </h3>
    </div>
    <div
      class="sc-fQpRED hGgTXR"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <p
        data-testid="employee-email"
      >
        luiz@caju.com.br
      </p>
    </div>
    <div
      class="sc-fQpRED hGgTXR"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <span
        data-testid="employee-admission-date"
      >
        22/10/2023
      </span>
    </div>
    <div
      class="sc-dsLQwm jxMBlB"
    >
      <button
        bgcolor="#ff8858"
        class="sc-gJhJTp buQVwd"
      >
        Revisar novamente
      </button>
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</div>
`;
